src/journal-remote/journal-gatewayd.c

```diff
@@ -239,6 +239,10 @@ static ssize_t request_reader_entries(
                 m->size = (uint64_t) sz;
         }
 
+        if (m->tmp == NULL && m->follow) {
+                return (ssize_t) 0;
+        }
```

Comment:
Please don't use {} around a single statement (see CODING_STYLE).

The cast is not needed, 0 is already a signed integer, and the compiler will do the size promotion automatically if necessary.